Source/WebInspectorUI/UserInterface/Views/DOMTreeElement.js

```diff
@@ -1271,18 +1271,37 @@ WI.DOMTreeElement = class DOMTreeElement extends WI.TreeElement
             return;
         }
 
-        var treeOutline = this.treeOutline;
-        var wasExpanded = this.expanded;
+        let treeOutline = this.treeOutline;
+        let wasExpanded = this.expanded;
+        let [newTagName, ...attributes] = newText.trim().split(" ");
 
-        function changeTagNameCallback(error, nodeId)
-        {
+        let changeTagNameCallback = (error, nodeId) => {
             if (error || !nodeId) {
                 cancel();
                 return;
             }
 
-            var node = WI.domManager.nodeForId(nodeId);
+            let node = WI.domManager.nodeForId(nodeId);
+
+            if (attributes.length) {
+                let attributesAddedCallback = (error, nodeId) => {
+                    if (error) {
+                        cancel();
+                        return;
+                    }
+
+                    nodeUpdatedCallback(node);
+                }
+
+                const attributeNameToReplace = "";
+                node.setAttribute(attributeNameToReplace, attributes.map(attr => attr.trim()).join(" "), attributesAddedCallback);
```

Comment:
i wonder if we should update `WI.DOMNode.prototype.setAttribute` to have the `name` be part of an `options = {}` instead

Style: missing `(` `)`
Style: please don't abbreviate `attribute`

Reply:
> i wonder if we should update `WI.DOMNode.prototype.setAttribute` to have the `name` be part of an `options = {}` instead

I'm inclined to leave it as is. It follows the expected arguments order of the DOM `setAttribute()` method. And it's only a Web Inspector implementation detail that we can omit the attribute name in favor of letting the engine create new ones from the value string.

I'd much rather add a new `WI.DOMNode.prototype.setAttributesAsText()` that omits the `name` entirely for these use cases. We can also leave it as is. 🤷‍♂️